onnxruntime/test/framework/inference_session_test.cc

```diff
@@ -1,5 +1,7 @@
 // Copyright (c) Microsoft Corporation. All rights reserved.
 // Licensed under the MIT License.
+#undef USE_CUDA  // TODO: Cuda is a shared library, so can't call any Cuda provider methods directly from here
```

Comment:
>#undef USE_CUDA [](http://example.com/codeflow?start=0&length=15)

here and elsewhere, some tests are being disabled. is this absolutely necessary?

Reply:
(we discussed this outside of github, but answering here for history) Yep, I'm logging a work item to rewrite the tests to work across the shared library boundary, the existing ones all call directly into the code and it'll take a few more weeks to translate them.